lib/core/configurator.c

```diff
@@ -765,6 +795,8 @@ void h2o_configurator__init_core(h2o_globalconf_t *conf)
         h2o_configurator_define_command(&c->super, "setenv",
                                         H2O_CONFIGURATOR_FLAG_ALL_LEVELS | H2O_CONFIGURATOR_FLAG_EXPECT_MAPPING, on_config_setenv);
         h2o_configurator_define_command(&c->super, "unsetenv", H2O_CONFIGURATOR_FLAG_ALL_LEVELS, on_config_unsetenv);
+        h2o_configurator_define_command(&c->super, "server-name", H2O_CONFIGURATOR_FLAG_GLOBAL, on_config_server_name);
+        h2o_configurator_define_command(&c->super, "display-server-name", H2O_CONFIGURATOR_FLAG_GLOBAL, on_config_display_server_name);
```

Comment:
Please add `| H2O_CONFIGURATOR_FLAG_EXPECT_SCALAR` to the flags of the two directives, so that configuration error would be displayed before calling the callbacks if a sequence or a mapping is specified as the values.